crates/service/src/worker.rs

```diff
@@ -75,17 +73,54 @@ impl WorkerOperations for Worker {
             Entry::Vacant(o) => {
                 let index =
                     Instance::create(self.path.join("indexes").join(handle.to_string()), options)?;
+                index.start();
                 o.insert(index);
                 protect.maintain(&self.view);
                 Ok(())
             }
-            Entry::Occupied(_) => Err(CreateError::Exist),
+            // reindex
+            Entry::Occupied(o) => {
+                {
+                    let index = o.remove();
+                    protect.maintain(&self.view);
+                    index.stop();
+                    let tracker = index.wait();
+                    drop(index);
+                    loop {
```

Comment:
Can this located inside `index.wait()`? Now, `index.wait()` only do an Arc clone, not wait.

Reply:
It's impossible.